segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java

```diff
@@ -408,5 +445,11 @@ private void logError(String segment, Throwable u) {
         }
     }
 
+    @Override
+    public void close() {
+        connection.close();
+        transientSegmentNames.forEach(name -> store.deleteStreamSegment(name, TIMEOUT));
```

Comment:
Let's put a comment next to this line explaining that we are purposefully not awaiting the completion of this and that the Segment store will eventually delete unused transient segments after some time (at its discretion).